packages/functions/src/coordinator/CoordinatorBookAppointmentHandler.ts

```diff
@@ -0,0 +1,25 @@
+import { FunctionsApi, MANUAL_DONOR_ID } from "@zm-blood-components/common";
+import { bookAppointment } from "../common/BookAppointmentHelper";
+import * as CoordinatorDAL from "../dal/AdminDataAccessLayer";
+
+export default async function (
+  request: FunctionsApi.CoordinatorBookAppointmentRequest,
+  callerId: string
+): Promise<FunctionsApi.BookAppointmentResponse> {
+  if (request.donorId !== MANUAL_DONOR_ID) {
+    throw Error(`Coordinator booking for real donots is not supported yet`);
+  }
+
+  const coordinator = await CoordinatorDAL.getCoordinator(callerId);
+
+  if (!coordinator) {
```

Comment:
What about checking the coordinator has the right permissions for the appointment (i.e. belongs to the right hospital)?